datacite/rest_client.py

```diff
@@ -25,49 +25,35 @@
 class DataCiteRESTClient(object):
     """DataCite REST API client wrapper."""
 
-    def __init__(self, username=None, password=None, url=None, prefix=None,
-                 test_mode=False, api_ver="2", timeout=None):
+    def __init__(self, username, password, url, prefix, timeout=None):
         """Initialize the REST client wrapper.
 
         :param username: DataCite username.
         :param password: DataCite password.
-        :param url: DataCite API base URL. Defaults to
-            https://api.datacite.org/.
-        :param test_mode: Set to True to change base url to
-        https://api.test.datacite.prg. Defaults to False.
+        :param url: DataCite API base URL.
         :param prefix: DOI prefix (or CFG_DATACITE_DOI_PREFIX).
-        :param api_ver: DataCite API version. Currently has no effect.
-            Default to 2.
         :param timeout: Connect and read timeout in seconds. Specify a tuple
             (connect, read) to specify each timeout individually.
         """
         self.username = username
         self.password = password
         self.prefix = prefix
-        self.api_ver = api_ver  # Currently not used
-
-        if test_mode:
-            self.api_url = 'https://api.test.datacite.org/'
-        else:
-            self.api_url = url or 'https://api.datacite.org/'
```

Comment:
I prefer to not require the user to provide the api url. In 99% of cases it's going to be the same. You could put it as a default in the function definition if you like that better.

Reply:
Indeed... but in my experience, developers go fast and test without pay attention. My fear is that hardcoding the production URL might create DOIs by mistake.
I can put, as default, the test one, what do you think?

Reply:
Double checking the code, I did not do it because of the constructor signature:
```
def __init__(self, username, password, url, prefix, timeout=None):
```

I would have to change the order of the params to:
```
def __init__(self, username, password, prefix, url="https://api.test.datacite.org/", timeout=None):
```
because `username`, `password` and `prefix` are **required**, while `url` can be optional.
That's why I did not change it in the end and set a default value.

Reply:
I do agree with Nico on the "possible mistake DOIs creation". Another possibility would be to default `test_mode` to `True`... That way is not changed, and the dev/used doesn't need to input the URL. OTH, it needs to pass the test mode param... So we are back to passing a param in any case.

Reply:
Oh...the constructor signature. Why couldn't they have put prefix first?

How about we split the difference. We can keep your setup with the url required for MDS, and switch the order in rest (since it hasn't been released)

```
class DataCiteRESTClient(object):
    """DataCite REST API client wrapper."""

    def __init__(self, username, password, prefix, test_mode=True, url=None, timeout=None):
```